numpy/core/src/multiarray/convert.c

```diff
@@ -350,16 +350,31 @@ PyArray_FillWithScalar(PyArrayObject *arr, PyObject *obj)
     }
     /* Python integer */
     else if (PyLong_Check(obj) || PyInt_Check(obj)) {
+        /* First attempt long long */
         npy_longlong v = PyLong_AsLongLong(obj);
         if (v == -1 && PyErr_Occurred()) {
-            return -1;
+            /* Long long failed, try unsigned long long */
```

Comment:
I'm not particularly happy with the nesting here, but this case is slightly different from the others in that we're mapping from a single Python type to one of several NumPy types. Suggestions welcome.